src/atomate2/cp2k/jobs/core.py

```diff
@@ -0,0 +1,341 @@
+    transformation_params : tuple of dict or None
+        The parameters used to instantiate each transformation class. Given as a list of
+        dicts.
+    input_set_generator : StaticSetGenerator
+        A generator used to make the input set.
+    write_input_set_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.write_cp2k_input_set`.
+    copy_cp2k_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.copy_cp2k_outputs`.
+    run_cp2k_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.run_cp2k`.
+    task_document_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.TaskDocument.from_directory`.
+    stop_children_kwargs : dict
+        Keyword arguments that will get passed to :obj:`.should_stop_children`.
+    write_additional_data : dict
+        Additional data to write to the current directory. Given as a dict of
+        {filename: data}. Note that if using FireWorks, dictionary keys cannot contain
+        the "." character which is typically used to denote file extensions. To avoid
+        this, use the ":" character, which will automatically be converted to ".". E.g.
+        ``{"my_file:txt": "contents of the file"}``.
+    """
+
+    name: str = "transmuter"
+    transformations: tuple[str, ...] = field(default_factory=tuple)
+    transformation_params: tuple[dict, ...] | None = None
+    input_set_generator: Cp2kInputGenerator = field(default_factory=StaticSetGenerator)
+
+    @cp2k_job
+    def make(
+        self,
+        structure: Structure,
+        prev_cp2k_dir: str | Path | None = None,
+    ):
+        """
+        Run a transmuter Cp2k job.
+
+        Parameters
+        ----------
+        structure : Structure
+            A pymatgen structure object.
+        prev_vasp_dir : str or Path or None
+            A previous Cp2k calculation directory to copy output files from.
+        """
+        transformations = _get_transformations(
+            self.transformations, self.transformation_params
+        )
+        ts = TransformedStructure(structure)
+        transmuter = StandardTransmuter([ts], transformations)
+        structure = transmuter.transformed_structures[-1].final_structure
+
+        # to avoid mongoDB errors, ":" is automatically converted to "."
+        if "transformations:json" not in self.write_additional_data:
+            tjson = transmuter.transformed_structures[-1]
+            self.write_additional_data["transformations:json"] = tjson
+
+        return super().make.original(self, structure, prev_cp2k_dir)
+
+@dataclass
+class MDMaker(BaseCp2kMaker):
+
+    name: str = "md"
+    input_set_generator: Cp2kInputGenerator = field(default_factory=MDSetGenerator)
+    task_document_kwargs: dict = field(
+        default_factory=lambda: {
+            "store_trajectory": True
+        }
+    )
+
+# TODO This should go in common
+def _get_transformations(
```

Comment:
Please move this to common and integrate it with the vasp version of the function.

Reply:
Moved to new `atomate2.common.utils` and updated vasp.